src/pyFAI/io/diffmap_config.py

```diff
@@ -0,0 +1,347 @@
+                elif "as_str" in methods:
+                    dico[key] = value.as_str()
+                elif "_asdict" in methods:   # namedtuple
+                    dico[key] = tuple(value)
+                else:
+                    dico[key] = value
+            else:
+                dico[key] = value
+        return dico
+
+    @classmethod
+    def from_dict(cls, dico, inplace=False):
+        """Alternative constructor,
+            * Normalize the dico (i.e. upgrade to the most recent version)
+            * accepts everything which is in OPTIONAL
+
+        :param dico: dict with the config
+        :param in-place: modify the dico in place ?
+        :return: instance of the dataclass
+        """
+        if not inplace:
+            dico = copy.copy(dico)
+
+        to_init = {}
+        for field in fields(cls):
+            key = field.name
+            if key in dico:
+                value = dico.pop(key)
+                if key in cls.ENFORCED:
+                    "Enforce a specific class type"
+                    klass = field.type
+                    if value is None:
+                        to_init[key] = value
+                    elif isinstance(value, (list, tuple)):
+                        if "from_serialized" in dir(klass):
+                            to_init[key] = klass.from_serialized(value)
+                        else:
+                            to_init[key] = klass(*value)
+                    elif isinstance(value, klass):
+                        to_init[key] = value
+                    elif isinstance(value, dict):
+                        to_init[key] = klass(**value)
+                    else:
+                        logger.warning(f"Unable to construct class {klass} with input {value} for key {key} in WorkerConfig.from_dict()")
+                        to_init[key] = value
+                else:
+                    to_init[key] = value
+        self = cls(**to_init)
+
+        for key in cls.GUESSED:
+            if key in dico:
+                dico.pop(key)
+        for key in cls.OPTIONAL+list(cls.DEPRECATED.keys()):
+            if key in dico:
+                value = dico.pop(key)
+                self.__setattr__(key, value)
+
+        if len(dico):
+            logger.warning("Those are the parameters which have not been converted !" + "\n".join(f"{key}: {val}" for key, val in dico.items()))
+        return self
+
+    def save(self, filename):
+        """Dump the content of the dataclass as JSON file"""
+        with open(filename, "w") as w:
+            w.write(json.dumps(self.as_dict(), indent=2))
+
+    @classmethod
+    def from_file(cls, filename: str):
+        """load the content of a JSON file and provide a dataclass instance"""
+        with open(filename, "r") as f:
+            dico = json.loads(f.read())
+        return cls.from_dict(dico, inplace=True)
+
+    # Compatibility layer !
+    @property
+    def npt_fast(self):
```

Comment:
Are there deprecation warnings raised by these ?

Reply:
Not yet ... but it is the place where to put them.